app/src/main/java/ch/epfl/sweng/studdybuddy/auth/GoogleSignInActivity.java

```diff
@@ -60,7 +67,20 @@ protected void onStart() {
             String personName = acct.getDisplayName();
             //appears only when the user is connected
             Toast.makeText(this, "Welcome " + personName, Toast.LENGTH_SHORT).show();
-            fetchUserAndStart(acct, NavigationActivity.class);
+            new Thread(new Runnable() {
```

Comment:
Could be good to abstract away the Runnable in the activity's controller

Reply:
Good suggestion, it will be added shortly